### PR TITLE
Add better error message if all features are dropped from dataset

### PR DESCRIPTION
**Pull request by tongyu-microsoft**
This PR adds better error message if all features are dropped. 
(responsibleai.exceptions.UserConfigValidationException: There is no feature in the dataset, or all features have been dropped from the dataset ) 

## Description
We added dropped_features where users can choose to drop certain features from the model. In the case when user drops all features from the dataset, we need to have detailed error message "There is no feature in the dataset, or all features have been dropped from the dataset"

## Checklist

- [ ] I have added screenshots above for all UI changes.
- [ ] I have added e2e tests for all UI changes.
- [ ] Documentation was updated if it was needed.

### CODE REVIEW

File: responsibleai/responsibleai/rai_insights/rai_insights.py [hunk 1]
```diff
@@ -471,6 +471,12 @@ def _validate_rai_insights_input_parameters(
                     columns=[target_column], axis=1)
                 small_test_data = small_test_data.drop(
                     columns=[target_column], axis=1)
+                if len(small_train_data.columns) == 0 or \
+                        len(small_test_data.columns) == 0:
+                    raise UserConfigValidationException(
+                        'There is no feature in the dataset,'
+                        ' or all features have been dropped from the dataset'
+                    )
```
Comment (gaugup) [inline responsibleai/responsibleai/rai_insights/rai_insights.py:479]: Can you check for specific condition and raise different exceptions? Raise one exception for no features and one for all features being part of dropped features?

**Review by gaugup — CHANGES_REQUESTED** [review]

File: responsibleai/responsibleai/rai_insights/rai_insights.py (hunk 1 above)
Comment (tongyu-microsoft) [reply responsibleai/responsibleai/rai_insights/rai_insights.py]: Updated!

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py [hunk 2]
```diff
@@ -377,6 +378,52 @@ def test_classes_exceptions(self):
...
+        with pytest.raises(UserConfigValidationException) as ucve:
+            RAIInsights(
+                model=model,
+                train=X_train,
+                test=X_test,
+                target_column=TARGET,
+                task_type='classification',
+                classes=[0, 1],
+                feature_metadata=FeatureMetadata(dropped_features=[
+                    "mean radius",
```
Comment (gaugup) [inline responsibleai/tests/rai_insights/test_rai_insights_validations.py:393]: you could just pass train.columns.tolist() here instead of the entire list of actual features here.

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py [hunk 3]
```diff
@@ -377,6 +378,52 @@ def test_classes_exceptions(self):
...
+                    "worst radius",
+                    "worst texture",
+                    "worst perimeter",
+                    "worst area",
+                    "worst smoothness",
+                    "worst compactness",
+                    "worst concavity",
+                    "worst concave points",
+                    "worst symmetry",
+                    "worst fractal dimension",
```
Comment (gaugup) [inline responsibleai/tests/rai_insights/test_rai_insights_validations.py:422]: Could you also add a test when x_train and x_test are empty that is they don't have any features?

File: responsibleai/responsibleai/rai_insights/rai_insights.py [hunk 4]
```diff
@@ -471,6 +473,16 @@ def _validate_rai_insights_input_parameters(
...
+                if len(small_train_data.columns) == 0 or \
+                        len(small_test_data.columns) == 0:
+                    if has_dropped_features:
+                        raise UserConfigValidationException(
+                            "All features have been dropped from the dataset"
+                        )
+                    else:
+                        raise UserConfigValidationException(
+                            "There is no feature in the dataset"
+                        )
```
Comment (gaugup) [inline responsibleai/responsibleai/rai_insights/rai_insights.py:485]: This should be actionable telling user what they need to do to avoid running into the exception again. Maybe add another blurb saying that "Please make sure that you dataset has at least one feature."?

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py (hunk 2 above)
Comment (tongyu-microsoft) [reply responsibleai/tests/rai_insights/test_rai_insights_validations.py:393]: updated!

File: responsibleai/responsibleai/rai_insights/rai_insights.py (hunk 4 above)
Comment (tongyu-microsoft) [reply responsibleai/responsibleai/rai_insights/rai_insights.py:485]: updated

File: responsibleai/responsibleai/rai_insights/rai_insights.py [hunk 5]
```diff
@@ -471,6 +473,16 @@ def _validate_rai_insights_input_parameters(
                     columns=[target_column], axis=1)
                 small_test_data = small_test_data.drop(
                     columns=[target_column], axis=1)
+                if len(small_train_data.columns) == 0 or \
+                        len(small_test_data.columns) == 0:
+                    if has_dropped_features:
+                        raise UserConfigValidationException(
+                            "All features have been dropped from the dataset"
```
[the remark this answers is not in the document]
Comment (tongyu-microsoft) [reply responsibleai/responsibleai/rai_insights/rai_insights.py:480]: added

File: responsibleai/responsibleai/rai_insights/rai_insights.py [hunk 6]
```diff
@@ -471,6 +473,19 @@ def _validate_rai_insights_input_parameters(
                     columns=[target_column], axis=1)
                 small_test_data = small_test_data.drop(
                     columns=[target_column], axis=1)
+                if len(small_train_data.columns) == 0 or \
+                        len(small_test_data.columns) == 0:
+                    if has_dropped_features:
+                        raise UserConfigValidationException(
+                            'All features have been dropped from the dataset.'
+                            ' Please do not drop all the features'
```
Comment (gaugup) [inline responsibleai/responsibleai/rai_insights/rai_insights.py:481]: (nit) missing `.` at the end of the sentence.

File: responsibleai/responsibleai/rai_insights/rai_insights.py [hunk 7]
```diff
@@ -471,6 +473,19 @@ def _validate_rai_insights_input_parameters(
...
+                        )
+                    else:
+                        raise UserConfigValidationException(
+                            'There is no feature in the dataset. Please make '
+                            'sure that your dataset contains at least '
+                            'one feature'
```
Comment (gaugup) [inline responsibleai/responsibleai/rai_insights/rai_insights.py:487]: (nit) missing `.` at the end of the sentence.

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py [hunk 8]
```diff
@@ -377,6 +378,52 @@ def test_classes_exceptions(self):
         assert 'The train labels and distinct values in ' + \
             'target (train data) do not match' in str(ucve.value)
 
+        X_train[TARGET] = y_train
```
[the remark this answers is not in the document]
Comment (gaugup) [reply responsibleai/tests/rai_insights/test_rai_insights_validations.py:382]: Somehow, I don't see another test added. Looks like we are adding more test checks in test_classes_exception()

File: responsibleai/responsibleai/rai_insights/rai_insights.py (hunk 7 above)
Comment (tongyu-microsoft) [reply responsibleai/responsibleai/rai_insights/rai_insights.py]: added

File: responsibleai/responsibleai/rai_insights/rai_insights.py (hunk 6 above)
Comment (tongyu-microsoft) [reply responsibleai/responsibleai/rai_insights/rai_insights.py]: added

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (tongyu-microsoft) [reply responsibleai/tests/rai_insights/test_rai_insights_validations.py]: sorry I misunderstood this comment. Now I added a test called test_dataset_exception

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py [hunk 9]
```diff
@@ -392,6 +394,19 @@ def test_classes_exceptions(self):
...
+                model=model,
+                train=X_train,
+                test=X_test,
+                target_column=TARGET,
+                task_type='classification',
+                classes=[0, 1],
+                feature_metadata=FeatureMetadata(
+                    dropped_features=X_train_feature_names))
+        assert 'The train labels and distinct values in target ' + \
+            '(train data) do not match' in str(ucve.value)
```
Comment (gaugup) [inline responsibleai/tests/rai_insights/test_rai_insights_validations.py:408]: Is the above validation needed or is it just copy/paste error? There seems to be the same assert in line 394-395.

**Review by gaugup — APPROVED** [review]

File: responsibleai/tests/rai_insights/test_rai_insights_validations.py (hunk 9 above)
Comment (tongyu-microsoft) [reply responsibleai/tests/rai_insights/test_rai_insights_validations.py:408]: that was to test when all features are dropped while feature class value is not matched. This can be removed. Removing
